### PR TITLE
Ref. Fix for issue #13561

### PR DESCRIPTION
**Pull request by belisarh**
Remove the trailing separator called in fullUrlWithQuery:

https://github.com/laravel/framework/issues/13561

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Our tests indicate that this is the expected behaviour.

**Comment by belisarh** [comment]
Yes tests should be updated. That should not be the expected behavior. 
The tests assertion is that when you add a parameter, the framework should for good measure append a path separator from `search?query=something` to `search/?query=something&page=5`

Using a CORS middleware for example would require routes to be written with a trailing slash if that even would work. Otherwise that extra "/" makes the middle ware not work on the route (any middle ware, think auth).
 
I have updated the tests and will update on this or a new pr. That is why Travis failed.

**Comment by belisarh** [comment]
Ah OK checked your comment on the issue. Thank you.

Will not update. However, heads up, middlewares do not work after you use the generated string on a subsequent request because of the intended behavior you got there. Chrome would strip the separator in the broweser, but sending it via ajax it is not stripped and left to be processed by the app. The app does not strip, does not match it to the "search" route and consequently does not apply any middleware you might have there.

**Comment by GrahamCampbell** [comment]
> Yes tests should be updated. That should not be the expected behavior. 

It is the expected behaviour though., That's we we explicitly added those tests to make sure someone doesn't change it.

**Comment by ifeltsweet** [comment]
Just because you have a test does not mean it is a correct test. You shouldn't be dismissing peoples concerns this easily.

We've just run into a similar same issue as @belisarh. There are inconsistencies in the tests. I'll explain why.

```
$request = Request::create('https://foo.com', 'GET');
$this->assertEquals('https://foo.com?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
    
$request = Request::create('https://foo.com?a=b', 'GET');
$this->assertEquals('https://foo.com/?a=b', $request->fullUrl());
```

See how one is returning `/?` and the other just `?`. According to [this commit](https://github.com/laravel/framework/commit/bc75f507d6de6a2ee7624f43bbf3a240688a5207) it should not be happening. Only the later one is correct. That's the first inconsistency.

Now look at this test:

```
$request = Request::create('http://foo.com/foo/bar?name=taylor', 'GET');
$this->assertEquals('http://foo.com/foo/bar?name=taylor', $request->fullUrl());
```

See how it's not adding `/`. That is correct. Now tell me why on earth will the following add a `/` (not in your tests, but that's how it works currently):

```
$request = Request::create('http://foo.com/foo/bar?name=taylor', 'GET');
$this->assertEquals('http://foo.com/foo/bar/?name=graham', $request->fullUrlWithQuery(['name' => 'graham']));
```

Why did it add a slash for me when I only asked it to change `name` query parameter. By convention Laravel actually strips slashes but in this case it adds one.

If you say that's expected behaviour then tell me why the following doesn't add slash for me:

```
$request = Request::create('http://foo.com/foo/bar', 'GET');
$this->assertEquals('http://foo.com/foo/bar?name=taylor', $request->fullUrlWithQuery(['name' => 'taylor']));
```

The bottom line is, @neochief made a correct commit adding `/` after domain name when there is nothing, but query. After that a few commits were made in wrong assumption that forward slash has to be added in other cases too. And it was made in inconsistent manner.

I believe I have it fixed in this PR https://github.com/laravel/framework/pull/13971
